### DISCUSSION TITLE
why is the new wfreerdp.dll only 122kb compared to our old 2000kb one

### DISCUSSION THREAD

**Discussion by lupuleasa-core** [FreeRDP/FreeRDP#8058] [Q&A]
We updated freerdp to a latest version, and we had crashes, with the dumpbin tool we noticed many things are missing from the wfreerdp.dll (like the freerdp_new function)

Anything happened with recent versions? We are updating a legacy product. Is it no longer backwards compatible?

we have updated the dll and the header files but the structure of the project seems refactored. are all things not contained anymore in the wfreerdp.dll?

EDIT: We noticed the interfaces have been moved to freerdp2.dll

**Comment by lupuleasa-core** [comment]
or an easier question:

what is the proper way to build wfreerdp.dll?
our legacy app is using the .dll version, not the .exe version

We are wondering if we made a mistake on the build procedure.

**Reply by akallabeth** [reply]
well, for a start it would be interesting in how you build your version. 

* do you need all in one dll?
* do you need client only or server components as well?

**Reply by lupuleasa-core** [reply]
oops, I have replied at top level
please see our comment here
https://github.com/FreeRDP/FreeRDP/discussions/8058#discussioncomment-3107602

**Reply by lupuleasa-core** [reply]
please let us know if there is a way to build the client into a **single** wfreerdp.dll, as I am not finding it in the docs and it would be of great help to us

**Comment by lupuleasa-core** [comment]
Hi @akallabeth 
Is it possible to build it all in one wfreerdp.dll?

**Reply by mfleisz** [reply]
@lupuleasa-core use `-DBUILTIN_CHANNELS=ON` to include all channels in the freerdp dll and `-DWITH_SERVER=OFF` to omit building server parts (since you don't seem to use them).

There are no build options currently available to build winpr/freerdp/freerdp-client into one big binary.

**Reply by lupuleasa-core** [reply]
thank you much

I have adapted our build command
```
cmake -GNinja -B./build -H./ -DCMAKE_INSTALL_PREFIX=C:\dev\freerdp -DCMAKE_BUILD_TYPE=Release -DCHANNEL_URBDRC=OFF -DBUILTIN_CHANNELS=ON -DWITH_SERVER=OFF
cmake --build build --target install
```

sadly the resulting wfreerdp.dll is still 123kb instead of 2,000kb, so we will look into upgrading our product to load and use all the new binaries available
